src/commands.ts

```diff
@@ -191,42 +191,45 @@ export async function approveToken(spender: string, amount: string) {
     this.resume();
 }
 
-export async function getTxParts(amount: string, fee: string, requestAdditional: string) {
-    let amounts: bigint[] = [];
-    amounts.push(await this.account.humanToShielded(amount));
-    if (requestAdditional == '+' || fee == '+') {
-        const additionalAmounts: string = await this.read('Enter additional space separated amounts (e.g. \'^1 ^2.34 ^50\'): ');
-        let convertedAmounts: bigint[] = await Promise.all(additionalAmounts.trim().split(/\s+/).map(async add => await this.account.humanToShielded(add)));
-        amounts = amounts.concat(convertedAmounts);
-    }
-
-    let actualFee: bigint;
-    if (fee === undefined || fee == '+') {
-        actualFee = await this.account.minFee();
-    } else {
-        actualFee = await this.account.humanToShielded(fee);
-    }
+export async function getTxParts(...amounts: string[]) {
+    const amountsBN: bigint[] = await Promise.all(amounts.map(amount => this.account.humanToShielded(amount)));
 
+    let entered = '';
+    let txType = TxType.Transfer;
+    this.echo(`[[;yellow;]Fee for transfer and withdraw transactions may vary]`)
+    this.resume();
+    do {
+        entered = (await this.read('Please specify tx type ([t]ransfer(default) or [w]ithdraw): ')).toLowerCase();
+        if (entered == 't' || entered == 'transfer') {
+            txType = TxType.Transfer;
+            break;
+        } else if (entered == 'w' || entered == 'withdraw') {
+            txType = TxType.Withdraw;
+            break;
+        }
+    } while(entered != '');
+    const txName = txType == TxType.Transfer ? 'transfer' : 'withdraw';
+    
     this.pause();
-    const result: TransferConfig[] = await this.account.getTxParts(amounts, actualFee);
+    const result: TransferConfig[] = await this.account.getTxParts(txType, amountsBN);
     this.resume();
 
     if (amounts.length > 1) {
-        this.echo(`Multi-destination request: ${ amounts.map(async a => `^${await this.account.shieldedToHuman(a)}`).join(', ') }`);
+        this.echo(`Multi-destination request: ${ amountsBN.map(async a => `^${await this.account.shieldedToHuman(a)}`).join(', ') }`);
```

Comment:
It seems that we don't await promises here:
![image](https://github.com/zkBob/zkbob-console/assets/17165678/2de4df0c-e3ea-4977-bf2d-50209d70b423)

Reply:
Fixed in https://github.com/zkBob/zkbob-console/pull/82/commits/eabc4ea94ca520fe321b4e25e7d15d8553a79162